client/js/analysis/gdpr.js

```diff
@@ -67,14 +79,117 @@ export async function getStreamingData(data) {
             Promise
                 .all(contentPromises)
                 .then((contentWithIndices) => {
-                  const content = contentWithIndices
-                      .map((entry) => entry.data)
-                      .reduce((dataA, dataB) => dataA.concat(dataB), []);
-                  resolve(content);
+                  resolve(
+                      contentWithIndices
+                          .map((entry) => entry.data)
+                          .reduce((dataA, dataB) => dataA.concat(dataB), [])
+                          .map((entry) => {
+                            // dates are strings in JSON, convert to JS date
+                            const endTime = new Date(entry.endTime);
+
+                            // move according to UTC offset
+                            const endTimeMs = +endTime;
+                            const endTimeOffset =
+                              endTime.getTimezoneOffset() * 60000;
+
+                            entry.endTime = new Date(endTimeMs - endTimeOffset);
+                            return entry;
+                          }),
+                  );
                 });
           });
         },
         (error) => reject(error),
     );
   });
 }
+
+/** @typedef {Map<string, Map<string, number>>} ArtistTrackTimeMap */
+/** @typedef {{
+ * start: Date,
+ * totals: ArtistTrackTimeMap
+ * }} ArtistTrackTimeWindow */
+
+/**
+ * Calculates aggregates from a list of GDPR records.
```

Comment:
Be more specific in this sentence, aggregates of what? The typedefs and return value give those details in a technical sense, but this comment should still give a reader enough detail to understand what this function does without having to piece together the values of the nested maps into a summary.